tests/unit/browser/test_caret.py

```diff
@@ -405,3 +395,82 @@ def test_reverse(self, caret, selection):
         caret.reverse_selection()
         caret.move_to_start_of_line()
         selection.check("one two three")
+
+
+class TestLineSelection:
+
+    def test_line_selection_toggle(self, caret, selection):
+        selection.toggle_line()
+        selection.check("one two three")
+
+    def test_line_selection_from_center(self, caret, selection):
+        caret.move_to_next_char(4)
+        selection.toggle_line()
+        selection.check("one two three")
+
+    def test_line_selection_more_lines(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_next_line(2)
+        selection.check_multiline("""
+                one two three
+                eins zwei drei
+
+                four five six
+                """, strip=True)
+
+    def test_line_selection_not_selecting_char(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_next_char()
+        selection.check("one two three")
+        caret.move_to_prev_char()
+        selection.check("one two three")
+
+    def test_line_selection_selecting_prev_next_word(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_next_word()
+        selection.check("one two three")
+        caret.move_to_prev_word()
+        selection.check("one two three")
+
+    def test_line_selection_selecting_end_word(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_end_of_word()
+        selection.check("one two three")
+
+    def test_line_selection_selecting_prev_next_line(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_next_line()
+        selection.check_multiline("""
+                one two three
+                eins zwei drei
+                """, strip=True)
+        caret.move_to_prev_line()
+        selection.check("one two three")
+
+    def test_line_selection_not_selecting_start_end_line(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_end_of_line()
+        selection.check("one two three")
+        caret.move_to_start_of_line()
+        selection.check("one two three")
+
+    def test_line_selection_selecting_block(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_end_of_next_block()
+        selection.check_multiline("""
+            one two three
+            eins zwei drei
+        """)
+
+    def test_line_selection_selecting_start_end_document(self, caret, selection):
+        selection.toggle_line()
+        caret.move_to_end_of_document()
+        selection.check_multiline("""
+                one two three
+                eins zwei drei
+
+                four five six
+                vier fünf sechs
+                """, strip=True)
```

Comment:
nitpick: Can you align this (and the others using `strip=True`) like the one on lines 460-463, so that the text is indented by one level, and the closing `"""` is on the same level as the function call?